src/uproot/interpretation/objects.py

```diff
@@ -122,6 +130,13 @@ def awkward_form(
         tobject_header=False,
         breadcrumbs=(),
     ):
+        awkward = uproot.extras.awkward()
+        if self._form is not None:  # TODO: is this really fine?
```

Comment:
@jpivarski i'm a bit uncomfortable with the piece of code i added here. The `_form` attribute already existed before, but was never directly returned here. The tests seem to pass, but do you know if there are potential problems that may be caused by directly returning `_form` if it is not None?

One thing that i had to do (and also don't quite feel comfortable with) was to convert the form from a dict representation which seems to happen sometimes (but i'm a bit unsure where and why not always).

Reply:
I'm still trying to understand your first paragraph, but about the second paragraph: there are three ways that a Form can be represented,

* as a `ak.forms.Form` object, which is immutable and therefore needs to be entirely known or at least built from the leaves up to the root
* as a Python dict (JSON represented as dicts and lists), which isn't type-safe, but it can be modified in place
* as a JSON string, which doesn't even ensure that pairs of square brackets are closed.

Thus, they have decreasing levels of safety and we prefer the safer ones when possible. The problem is that the process of discovering the type involves starting at the root and walking down toward the leaves, filling in a ListOffsetArray's content if we see a non-empty example of it. So the ideal, `ak.forms.Form`, isn't possible and the Form starts its life as a Python dict. When it gets converted into an object, that's final. The Forth is fully discovered at that point.

Reply:
i now realized that the `self._form` attribute will always be in the Python dict form (either from generated or known forth code). So at the place where it is returned in `AsObjects.awkward_form(...)` it has to be converted to the final `ak.forms.Form` object. In principle the form from the known forth path could have been provided directly as `ak.forms.Form`, but to be consistent with the one created from the ForthGenerator it is now also a dict.